dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task210816DeInodeRelationship.java

```diff
@@ -0,0 +1,109 @@
+package com.dotmarketing.startup.runonce;
+
+import com.dotmarketing.common.db.DotDatabaseMetaData;
+import com.dotmarketing.startup.AbstractJDBCStartupTask;
+import com.dotmarketing.util.Logger;
+import io.vavr.control.Try;
+import java.sql.SQLException;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * 1. Adds `mod_date` column to `relationship` table
+ * 2. Copies `inode.idate` to new `relationship.mod_date`
+ * 3. Drops FK from `relationship` to `inode`
+ */
+public class Task210816DeInodeRelationship extends AbstractJDBCStartupTask {
+
+    private final DotDatabaseMetaData dotDatabaseMetaData = new DotDatabaseMetaData();
+    private final String COPY_RELATIONSHIP_MOD_DATE_FROM_INODE =
+            "UPDATE relationship as r SET mod_date = i.idate FROM inode i WHERE r.inode = i.inode;";
+    private final String DELETE_RELATIONSHIPS_FROM_INODE_BY_TYPE = "DELETE FROM inode WHERE type = 'relationship';";
+    private final String DELETE_RELATIONSHIPS_FROM_INODE_BY_JOIN = "DELETE FROM inode where exists(select 1 from relationship r where r.inode = inode.inode);";
+
+
+    @Override
+    public boolean forceRun() {
+        try {
+            return !dotDatabaseMetaData.hasColumn("relationship", "mod_date")
+                    || Try.of(()->dotDatabaseMetaData.getConstraints("relationship")).
+                    getOrElse(Collections.emptyList())
+                    .stream().anyMatch("fkf06476385fb51eb"::equals);
```

Comment:
why do you need to match against a hard-coded fk id?

Reply:
that's the name we gave it to that constraint https://github.com/dotCMS/core/blob/42b2a6fca8fb7196a0224090f965c9cfbcf0228e/dotCMS/src/main/resources/postgres.sql#L1561

Reply:
I wonder what it is called in older datasets?  I feel like this will fail on some dbs.  Maybe delete all `FK`s and then rebuild the ones we still need?

Reply:
@wezell  It's been like this since 3.0 but can't tell before that... probably delete/rebuild is better

Reply:
@nollymar @wezell Done. Not relying on FK hardcoded name anymore.